### ISSUE TITLE
How to improve the keyoint detection accuracy of my YOLO model?

### ISSUE THREAD

**Issue by Jonathan-JB1** [ultralytics/ultralytics#25444]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

Hello all, I am looking for tips on how to improve my YOLO model. I trained a yolo11l-pose.pt model to detect coins and fire salamanders and 15 keypoints on the fire salamanders. The model works well but I want to improve the accuracy of the keypoints. However, nothing I tried seems to really work. 

I added 300 images to my original dataset of 500 images. At first, it seemingly made the model worse but after rechecking and correcting all annotations it showed minimal improvements. This confused me since most sources say a large dataset is the most impactful improvement one can do. 

I also tried increasing the image size (originally 640 x 640), the batch size (originally 16) and the number of epochs (originally 200) as well as changing the learning rate and using the patience parameter. Overall, no combination I tried was significantly better than the others. 

For my most recent model the pose mAP50 value is 0.995 and the mAP50-95 value is 0.878 which is already good but when using the model I often find images where the accuracy could be better.

From what I can gather there are a lot of other parameters and pre-processing steps that can influence the accuracy but I don’t really know where to start. I appreciate any tips and advice!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @Jonathan-JB1, thank you for your interest in Ultralytics 🚀! We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python) and [CLI](https://docs.ultralytics.com/usage/cli) usage examples and where many of the most common questions may already be answered.

For improving pose/keypoint accuracy, please provide as much information as possible about your dataset, annotations, training configuration, validation results, and representative failure cases. This will help identify whether the remaining errors are related to annotation quality, keypoint visibility, image resolution, pose variation, or dataset distribution. The [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips) may also be useful.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example) to help us debug it.

If this is a custom training ❓ Question, please provide as much information as possible, including dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips).

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

This is an automated response 🤖, and an Ultralytics engineer will also assist you soon.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo26"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes) and [Tasks](https://docs.ultralytics.com/tasks) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by Y-T-G** [comment]
Your dataset is still too small for deep learning.

You can try using `yolo26l.pt` for training.

Also make sure your images are unique. Adding augmented versions of the same image don't bring diversity to the dataset and increases chance of overfitting.

> For my most recent model the pose mAP50 value is 0.995 and the mAP50-95 value is 0.878 which is already good but when using the model I often find images where the accuracy could be better.

If your validation set only has few images, then the metrics you get wouldn't be reliable or statistically significant. You can easily fluke high scores on a small validation set.

**Comment by Jonathan-JB1** [comment]
Thanks for the reply! Roughly how large should my dataset be for deep learning? When adding new unique images, should I focus on cases where the model struggled to find the keypoints during my tests? Many of my current images have similar backgrounds and animal poses, so I assume adding more variety would benefit the model.

**Comment by ramelito** [comment]
From my experience dataset should contain at least 10k various images for models to start generalize, 10 percent for val and 20 percent for test

**Comment by glenn-jocher** [comment]
There is no universal 10k-image minimum; diversity and label consistency matter more than a fixed count. Yes, prioritize unique deployment-representative failure cases—unusual poses, backgrounds, lighting, scale, blur, and occlusion—while keeping a frozen test set and splitting by animal/video/session to prevent leakage. With only ~800 images, consider [k-fold cross-validation](https://docs.ultralytics.com/guides/kfold-cross-validation) for more reliable model comparisons.

**Comment by Jonathan-JB1** [comment]
Thank you both for your advice!

**Comment by glenn-jocher** [comment]
You’re welcome! Keep the validation/test split frozen and change one variable at a time so future comparisons remain meaningful.

**Comment by glenn-jocher** [comment]
Closing the answered training/data question following your acknowledgment. Preserve the [maintainer guidance](https://github.com/ultralytics/ultralytics/issues/25444#issuecomment-5111192559): there is no universal 10,000-image minimum. Compare changes on a frozen, deployment-representative test set with animal/video/session separation; add unique failure cases and check label consistency. High validation AP on a small or similar split does not establish deployment generalization. No reproducible package defect was identified here.
